wire-ios-sync-engine/Source/Synchronization/SyncAgent.swift

```diff
@@ -280,6 +289,20 @@ final class SyncAgent: NSObject, SyncAgentProtocol {
                 self?.resume()
             }
     }
+
+    private func handlePushChannelAlreadyOpened() {
+        pushChannelMonitor.startMonitoring { [weak self] in
+            self?.retrySyncAfterPushChannelClosed()
+        }
+        WireLogger.sync.debug("push channel already opened, request NSE release", attributes: .syncAttributes)
+        pushChannelMonitor.notify()
+    }
+
+    private func retrySyncAfterPushChannelClosed() {
+        WireLogger.sync.debug("retry sync after NSE push channel closed", attributes: .syncAttributes)
+        pushChannelMonitor.stopMonitoring()
+        resume()
+    }
```

Comment:
communication with NSE: If the main app opens, we want the NSE to release the lock immediately. For the user that way the app shows messages directly within the app